astroplan/exptime.py

```diff
@@ -0,0 +1,177 @@
+    (from pg 57-58 of [1]_).
+
+    Parameters
+    ----------
+    snr : float, int, or `~astropy.units.Quantity`
+        The signal to noise ratio of the given observation in dimensionless
+        units.
+    waveset : array-like
+        The wavelengths associated with the target's flux.
+    flux : array-like
+        The flux of the target.
+    observer : `~astroplan.observer.Observer`
+        The Observer object.
+    telescope : `~astroplan.telescope.Telescope`
+        The Telescope object.
+    npix : `~astropy.units.Quantity`, optional
+        Number of pixels under consideration for the signal with units of
+        pixels. Default is 1 * astropy.units.pixel.
+    n_background : `~astropy.units.Quantity`, optional
+        Number of pixels used in the background estimation with units of
+        pixels. Default is set to np.inf * astropy.units.pixel
+        such that there is no contribution of error due to background
+        estimation. This assumes that n_background will be >> npix.
+    background_rate : `~astropy.units.Quantity`, optional
+        Photons per pixel per second due to the backround/sky with units of
+        counts/second/pixel.
+        Default is 0 * (astropy.units.ct /
+        astropy.units.second / astropy.units.pixel)
+    darkcurrent_rate : `~astropy.units.Quantity`, optional
+        Counts per pixel per second due to the dark current with units
+        of counts/second/pixel.
+        Default is 0 * (astropy.units.ct /
+        astropy.units.second / astropy.units.pixel)
+
+    References
+    ----------
+    .. [1] Howell, S. B. 2000, *Handbook of CCD Astronomy* (Cambridge, UK:
+        Cambridge University Press)
+
+    Returns
+    -------
+    t : `~astropy.units.Quantity`
+        The exposure time needed (in seconds) to achieve the given signal
+        to noise ratio.
+    """
+    # import synphot, which isn't a required package of astroplan
+    from synphot.models import Empirical1D
+    from synphot.spectrum import SourceSpectrum, SpectralElement
+    from synphot.observation import Observation
+
+    # set the source spectrum with synphot
+    source_spec = SourceSpectrum(Empirical1D, points=waveset,
+                                 lookup_table=flux)
+
+    # set the spectral elements if given (quantum efficiency,skymodel,bandpass)
+    qe = telescope.ccd_response
+    skymodel = observer.skymodel
+
+    spec_elements = _get_spectral_element(telescope.bandpass,
+                                          SpectralElement, Empirical1D)
+    if skymodel is not False:
+        spec_elements *= _get_spectral_element(skymodel,
+                                               SpectralElement, Empirical1D)
+    if qe is not False:
+        spec_elements *= _get_spectral_element(qe,
+                                               SpectralElement, Empirical1D)
+
+    # set the forcing if the source spec and bandpass don't overlap
+    # hmm.. how to decide? will set to taper for now
```

Comment:
Is this comment still necessary?

Reply:
Ah, yeah, this is actually kind of an important distinction. The issue arises when the source spectrum and bandpass don't fully overlap - you can tell synphot to force the convolution anyway, either by making the incomplete spectral component taper to zero for its missing wavelengths, or by guessing/extrapolating what the remainder of the spectrum will be. Extrapolating might be okay if only a tiny bit of the spectrum is missing, but otherwise sounds a bit risky to me, which is why I chose to taper. 

All that said, I think the best approach would be to add an optional parameter like `force_overlap` to `exptime_from_ccd_snr()` with a default of `taper`, and perhaps raise a warning if this forcing is implemented (which synphot might already do, actually). That way the user could both be aware that they don't overlap and could decide for themselves how they want to handle it.